PennyPincher/PennyPincherAndroidGesturesImporter.swift

```diff
@@ -0,0 +1,200 @@
+        //    Gesture
+        //                8 bytes     long        Gesture ID
+        //                4 bytes     int         Number of strokes
+        //    Stroke
+        //                4 bytes     int         Number of points
+        //    Point
+        //                4 bytes     float       X coordinate of the point
+        //                4 bytes     float       Y coordinate of the point
+        //                8 bytes     long        Time stamp
+        //
+
+        let reader = BigEndianDataReader(data: data)
+        let version = reader.getInt(zeroedType: Int16())
+        let entryCount = reader.getInt(zeroedType: Int32())
+        if debug { print("Header data: version=\(version) entries=\(entryCount)") }
+        
+        guard version == 1 else {
+            print("PennyPincherAndroidGesturesImporter Error. Unsupported file version \(version)")
+            return []
+        }
+        
+        for entryNumber in 0..<entryCount {
+            //Entry
+            let unk = reader.getInt(zeroedType: Int16()) //there's 2 extra bytes here for some reason....
+            let entryName = reader.getCString()
+            let gestureCount = reader.getInt(zeroedType: Int32())
+            if debug { print("Entry #\(entryNumber): name=\(entryName) gestures=\(gestureCount) Unk=\(unk)") }
+            
+            for gestureNumber in 0..<gestureCount {
+                let gestureId = reader.getInt(zeroedType: Int64())
+                let strokeCount = reader.getInt(zeroedType: Int32())
+                if debug { print("Gesture #\(gestureNumber): id=\(gestureId) strokes=\(strokeCount)") }
+                
+                var strokes = [ImportedStrokes]()
+                for strokeNumber in 0..<strokeCount {
+                    
+                    let pointsCount = reader.getInt(zeroedType: Int32())
+
+                    if debug { print("Stroke #\(strokeNumber): points=\(pointsCount)") }
+                    
+                    var cgPoints = [CGPoint]()
+                    for pointNumber in 0..<pointsCount {
+                        let point = CGPoint(x: CGFloat(reader.getFloat32()), y: CGFloat(reader.getFloat32()))
+                        let timeStamp = reader.getInt(zeroedType: Int64())
+                        cgPoints.append(point)
+                        
+                        if debug { print("Point #\(pointNumber): (\(point.x), \(point.y)) timestamp:\(timeStamp)") }
+                    }
+                    strokes.append(ImportedStrokes(points: cgPoints).normalized())
+                }
+                
+                let importedGesture = ImportedGesture(id: entryName, strokes: strokes)
+                translatedGestures.append(importedGesture)
+            }
+        }
+        
+        return translatedGestures
+    }
+}
+
+private protocol ByteSwappable {
+    var byteSwapped: Self { get }
+    init()
+}
```

Comment:
Swapping endianness implies an assumption about the default endianness, which you might not have intended. Do you mean to require and use `.bigEndian` and `.littleEndian`, which are each no-ops depending on the architecture, instead of `.byteSwapped`?